### ISSUE TITLE
[Bug]: alter table add/drop primary key is not supported on TAE level

### ISSUE THREAD

**Issue by domingozhang** [matrixorigin/matrixone#8740]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

create table t1(a int, b int);
create table t2(a int primary key, b int);

alter table t1 add primary key(a,b);
ERROR 20101 (HY000): internal error: panic runtime error: invalid memory address or nil pointer dereference: 
runtime.panicmem
	/usr/local/go/src/runtime/panic.go:260
runtime.sigpanic
	/usr/local/go/src/runtime/signal_unix.go:835
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Scope).AlterTable
	/var/root/matrixone/pkg/sql/compile/ddl.go:169
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Compile).Run
	/var/root/matrixone/pkg/sql/compile/compile.go:191
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).

mysql> alter table t2 drop primary key;
Query OK, 0 rows affected (0.03 sec)

mysql> insert into t2 values(1,2),(1,3);
ERROR 1062 (HY000): Duplicate entry '1' for key 'a'

The primary key still exists.

**Comment by XuPeng-SH** [comment]
it depends on #8014

**Comment by XuPeng-SH** [comment]
subtask of #8014

**Comment by aptend** [comment]
designing

**Comment by aptend** [comment]
Postponed. Handle add/drop normal columns first

**Comment by matrix-meow** [comment]
Hello @domingozhang. The bug issue in the BVT test code has not been removed,issues automatically open.

**Comment by matrix-meow** [comment]
Hello @domingozhang. The bug issue in the BVT test code has not been removed,issues automatically open.

**Comment by domingozhang** [comment]
please help to close it @aressu1985

**Comment by matrix-meow** [comment]
Hello @domingozhang. The bug issue in the BVT test code has not been removed,issues automatically open.
